inflation/lp/InflationLP.py

```diff
@@ -0,0 +1,1890 @@
+        try:
+            return self.atomic_monomial_from_hash[key.tobytes()]
+        except KeyError:
+            if len(self.lexorder_symmetries) == 1:
+                mon = InternalAtomicMonomial(self, array1d)
+                self.atomic_monomial_from_hash[key.tobytes()] = mon
+                return mon
+            else:
+                mon_as_symboolvec = key[self.lexorder_symmetries]
+                mon_as_symboolvec = mon_as_symboolvec[
+                    np.lexsort(mon_as_symboolvec.T[::-1])]
+                mon_as_boolvec = mon_as_symboolvec[-1]
+                mon = InternalAtomicMonomial(self, 
+                                             np.flatnonzero(mon_as_boolvec))
+                for alt_key in mon_as_symboolvec:
+                    self.atomic_monomial_from_hash[alt_key.tobytes()] = mon
+                return mon
+
+    def Monomial(self, array1d: np.ndarray, idx: int = -1) -> CompoundMoment:
+        r"""Create an instance of the `CompoundMonomial` class from a 2D array.
+        An instance of `CompoundMonomial` is a collection of
+        `InternalAtomicMonomial`.
+
+        Parameters
+        ----------
+        array2d : numpy.ndarray
+            Moment encoded as a 1D array of integers, relative to _lexorder
+        idx : int, optional
+            Assigns an integer index to the resulting monomial, which can be
+            used as an id, by default -1.
+
+        Returns
+        -------
+        CompoundMoment
+            The monomial factorised into AtomicMonomials, all brought to
+            representative form under inflation symmetries.
+        """
+        _factors = self.factorize_monomial_1d(array1d, canonical_order=False)
+        list_of_atoms = [self._AtomicMonomial(factor)
+                         for factor in _factors if len(factor)]
+        mon = self._monomial_from_atoms(list_of_atoms)
+        mon.attach_idx(idx)
+        return mon
+
+    def _monomial_from_atoms(self,
+                             atoms: List[InternalAtomicMonomial]
+                             ) -> CompoundMoment:
+        """Build an instance of `CompoundMonomial` from a list of instances
+        of `InternalAtomicMonomial`.
+
+        Parameters
+        ----------
+        atoms : List[InternalAtomicMonomial]
+            List of instances of `InternalAtomicMonomial`.
+
+        Returns
+        -------
+        CompoundMoment
+            A `CompoundMonomial` with atomic factors given by `atoms`.
+        """
+        key = tuple(sorted(atoms))
+        try:
+            return self.monomial_from_atoms[key]
+        except KeyError:
+            mon = CompoundMoment(atoms)
+            try:
+                mon.idx = self.first_free_idx
+                self.first_free_idx += 1
+            except AttributeError:
+                pass
+            self.monomial_from_atoms[key] = mon
+            self.monomial_from_name[mon.name] = mon  # TODO: Add simple name and complex name
```

Comment:
Have we done this/does this need to be done?

Reply:
Added a new issue.